lease/lessor.go

```diff
@@ -614,8 +624,9 @@ func (l *Lease) Keys() []string {
 
 // Remaining returns the remaining time of the lease.
 func (l *Lease) Remaining() time.Duration {
-	t := monotime.Time(atomic.LoadUint64((*uint64)(&l.expiry)))
-	return time.Duration(t - monotime.Now())
+	l.expiryMu.RLock()
+	defer l.expiryMu.RUnlock()
+	return l.expiry.Sub(time.Now())
```

Comment:
since `forever` breaks down when switching to `time.Time`, this patch could instead use time.Time{} to mean forever:

```go
func (l *Lease) Remaining() time.Duration {
    l.expiryMu.RLock()
    defer l.expiryMu.RUnlock()
    if l.expiry.IsZero() {
        return time.Duration(math.MaxInt64)
    }
    return time.Until(l.expiry)
}
```
```